cmd/vela-worker/operate.go

```diff
@@ -43,13 +41,58 @@ func (w *Worker) operate(ctx context.Context) error {
 	token := <-w.RegisterToken
 
 	logrus.Trace("received register token")
-
+	logrus.Trace("setting up vela client")
 	// setup the vela client with the token
 	w.VelaClient, err = setupClient(w.Config.Server, token)
 	if err != nil {
 		return err
 	}
 
+	logrus.Trace("getting queue creds")
+
+	creds, r, err := w.VelaClient.Queue.GetInfo()
+	if err != nil {
+		logrus.Trace("error getting creds")
+		logrus.Errorf("unable to retrieve queue credentials with status code %v", r.StatusCode)
+		return err
+	}
+
+	logrus.Trace("wait for queue details before setup queue")
+
+	// once worker created/check in, queue details are used to setup queue here.
```

Comment:
```suggestion
	// set queue address and public key using credentials received from server
```
Worker technically hasn't checked in yet